extensions/pm2/src/components.tsx

```diff
@@ -0,0 +1,147 @@
+import { ActionPanel, Action, Color, Icon, List, confirmAlert } from "@raycast/api";
+import { getProgressIcon } from "@raycast/utils";
+import { formatDistance } from "date-fns";
+import { get } from "lodash";
+import { ProcessDescription } from "pm2";
+import { getProcessStatusColor, getRaycastIcon, isRaycastNodeProcess, runPm2Command } from "./utils.js";
+import { ExportedKey, Pm2Process } from "./types.js";
+
+export const MetaData = ({ processDescription: p }: { processDescription: ProcessDescription }) => {
+  const isRaycastProcess = isRaycastNodeProcess(p);
+  const exportedKeys: ExportedKey[] = [
+    { key: "name" },
+    { key: "pm_id", render: (v) => String(v) },
+    { key: "pid" },
+    { key: "pm2_env.namespace" },
+    { key: "pm2_env.version" },
+    { key: "pm2_env.exec_mode" },
+    {
+      key: "pm2_env._",
+      render: (v) => (isRaycastProcess ? "node" : String(v)),
+      icon: isRaycastProcess ? getRaycastIcon() : undefined,
+      title: "env._",
+    },
+    { key: "pm2_env.pm_pid_path" },
+    { key: "pm2_env.pm_exec_path" },
+    { key: "pm2_env.pm_uptime", render: (v) => formatDistance(new Date(v), new Date()) },
+    { key: "pm2_env.username" },
+    { key: "pm2_env.watch" },
+  ];
+
+  return (
+    <List.Item.Detail.Metadata>
+      {exportedKeys.map((item) => {
+        const key = typeof item === "string" ? item : item.key;
+        const value = get(p, key);
+        return (
+          value !== undefined && (
+            <List.Item.Detail.Metadata.Label
+              key={key}
+              icon={typeof item === "string" ? undefined : item.icon}
+              title={item.title ?? (key.split(".").at(-1) as string)}
+              text={item.render?.(value) ?? String(value)}
+            />
+          )
+        );
+      })}
+      <List.Item.Detail.Metadata.Separator />
+      <List.Item.Detail.Metadata.TagList title="status">
+        {typeof p.pm2_env?.status === "string" && (
+          <List.Item.Detail.Metadata.TagList.Item
+            text={p.pm2_env?.status}
+            color={getProcessStatusColor(p.pm2_env?.status)}
+          />
+        )}
+        {typeof p.monit?.cpu === "number" && (
```

Comment:
This code could be improved, instead of typing a lot of 
```
typeof p.pm2_env?.status === "string" && `
```

It's better to check for that earlier in the function, if you do that you don't need to duplicate your logic that you do here. I mean that's better code. (No AI involved just a dev reading code)

Reply:
early return pattern, guard close.. Let me show you an article.
https://medium.com/swlh/return-early-pattern-3d18a41bba8

That's not the best to explain there is some that are better, but you get the idea.

Reply:
actually when you always go for an early return pattern you will never look back. That is a lot better.

Reply:
@Elmona Hey, thanks. I know this. But no. I want to keep my code style consistent here for now. I plan to refactor this part of the code entirely, but I don't want to spend too much energy on so-called "best practices".

I prefer to find out if an ESLint rule could automatically check and format the code into a unified format. I don't want to spend energy on these ambiguous code styles.

In addition, if you are obsessed with code style, I recommend that you learn about [XO](https://github.com/xojs/xo) code style. That is what I expect.

Anyway, thanks for the advice.

Reply:
Fine, truth be told and most looked good, i'm not obsessed with any pattern but I always try to choose the best myself.

I don't mean this as an git blame more that here is a file you will improve on next time you type it ❤️